SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc

```diff
@@ -0,0 +1,241 @@
+    return theTotalResult;
+  }
+
+  // Go on to regular case
+  //
+  G4double originalEnergy = aParticle->GetKineticEnergy();
+  G4double kineticEnergy = originalEnergy;
+
+  // Get kinetic energy per nucleon for ions
+  if (aParticle->GetParticleDefinition()->GetBaryonNumber() > 1.5)
+    kineticEnergy /= aParticle->GetParticleDefinition()->GetBaryonNumber();
+
+  try {
+    theInteraction = GetHadronicInteractionList().at(0);
+    //    theInteraction = GetHadronicInteractionList()[0];
+    //      ChooseHadronicInteraction( kineticEnergy, aMaterial, anElement );
+  } catch (G4HadronicException& aE) {
+    G4ExceptionDescription ed;
+    aE.Report(ed);
+    ed << "Target element " << anElement->GetName() << "  Z= " << targetNucleus.GetZ_asInt()
+       << "  A= " << targetNucleus.GetA_asInt() << G4endl;
+    DumpState(aTrack, "ChooseHadronicInteraction", ed);
+    ed << " No HadronicInteraction found out" << G4endl;
+    G4Exception("G4SQInelasticProcess::PostStepDoIt", "had005", FatalException, ed);
+  }
+
+  // Initialize the hadronic projectile from the track
+  thePro.Initialise(aTrack);
+
+  G4HadFinalState* result = 0;
+  G4int reentryCount = 0;
+
+  do {
+    try {
+      // Call the interaction
+      result = theInteraction->ApplyYourself(thePro, targetNucleus);
+      ++reentryCount;
+    } catch (G4HadronicException& aR) {
+      G4ExceptionDescription ed;
+      aR.Report(ed);
+      ed << "Call for " << theInteraction->GetModelName() << G4endl;
+      ed << "Target element " << anElement->GetName() << "  Z= " << targetNucleus.GetZ_asInt()
+         << "  A= " << targetNucleus.GetA_asInt() << G4endl;
+      DumpState(aTrack, "ApplyYourself", ed);
+      ed << " ApplyYourself failed" << G4endl;
+      G4Exception("G4SQInelasticProcess::PostStepDoIt", "had006", FatalException, ed);
+    }
+
+    edm::LogInfo("G4SQInelasticProcess::PostStepDoIt") << "Call for " << theInteraction->GetModelName() << std::endl;
+    edm::LogInfo("G4SQInelasticProcess::PostStepDoIt")
+        << "Target element " << anElement->GetName() << "  Z=" << targetNucleus.GetZ_asInt()
+        << "  A=" << targetNucleus.GetA_asInt() << std::endl;
+
+    edm::LogInfo("G4SQInelasticProcess::PostStepDoIt")
+        << "Nr of secondaries: " << result->GetNumberOfSecondaries() << std::endl
+        << "Momentum change and E deeposit: " << result->GetMomentumChange() << " " << result->GetLocalEnergyDeposit()
+        << std::endl
+        << "Track position and vertex: " << aTrack.GetPosition() << " " << aTrack.GetVertexPosition() << std::endl;
+
+    float r = aTrack.GetPosition().perp();
+    float z = fabs(aTrack.GetPosition().z());
+    edm::LogInfo("G4SQInelasticProcess::PostStepDoIt")
+        << "In tracker volume? " << (r < (100 * cm) && z < (200 * cm) ? " YES!  " : " NO!  ") << "r=" << r / cm
+        << " z=" << z / cm << std::endl;
+
+    // Check the result for catastrophic energy non-conservation
```

Comment:
Again: it is enough to make 1 check in your case. LogInfo -> LogVerbatim

Reply:
Ok, anyway whole method removed as per your later comment.